ProcessMaker/processmaker
# Enhanced Project Filtering and Parameters Implementation

## Issue & Reproduction Steps

This PR addresses the need for Project filtering within the application. It introduces project parameters for project members, categories, and projects, enhancing the capabilities of the PMQL Filter component.

## Solution
In this PR, the following enhancements and features have been added to facilitate project filtering:

- Introduction of the `searchProjects` and `searchProjectAll`, `buildProjectsPmql` methods to enable project filtering.

## How to Test

1. Ensure you have the Projects package installed on the task/FOUR-10028 branch.
2. Confirm that Projects have been created in the system.
3. Navigate to the Projects listing page.
4. Click the 'Filter' button to apply various filtering options.

**Expected Behavior**
Filtering projects using the new options should not result in any errors, and relevant data should be displayed correctly in the data table.

## Related Tickets & Packages
- Ticket [FOUR-10028](https://processmaker.atlassian.net/browse/FOUR-10028)
- [Projects Package PR](https://github.com/ProcessMaker/package-projects/pull/33)

ci:next
ci:package-projects:task/FOUR-10028

## Code Review Checklist


[FOUR-10028]: https://processmaker.atlassian.net/browse/FOUR-10028?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
